supabase/functions/updates/index.ts

```diff
@@ -25,6 +25,33 @@ const headersSchema = z.object({
 
 const bypassRedis = false
 
+async function incrementDeviceRequestCount(ipAddress: string): Promise<void> {
+  const redis = await getRedis();
+  if (redis) {
+    const deviceIpHashKey = `device_requests:${createHash("md5").update(ipAddress).toString()}`;
+    await redis.hincrby(deviceIpHashKey, 'count', 1);
+  }
+}
+
+async function getDeviceRequestCount(ipAddress: string): Promise<number | undefined> {
+  const redis = await getRedis();
+  if (redis) {
+    const deviceIpHashKey = `device_requests:${createHash("md5").update(ipAddress).toString()}`;
+    const count = await redis.hget(deviceIpHashKey, 'count');
+    if (count !== null) {
+      return Number(count);
+    }
+  }
+}
+
+async function blacklistDeviceRequest(ipAddress: string): Promise<void> {
```

Comment:
Can we use another term than blacklist? This one has an implicit racial connotation, you can read more about it [here](https://www.splunk.com/en_us/blog/learn/blacklist-whitelist-inclusivity.html#:~:text=The%20term%20%E2%80%9Cblacklist%E2%80%9D%20contains%20the,and%20contribute%20to%20systematic%20racism.).

We could use block list instead.